deploy/kubernetes/helm/che/templates/configmap.yaml

```diff
@@ -63,6 +63,19 @@ data:
   CHE_INFRA_KUBERNETES_PVC_QUANTITY: {{ .Values.global.pvcClaim }}
   CHE_INFRA_KUBERNETES_PVC_PRECREATE__SUBPATHS: "true"
   CHE_INFRA_KUBERNETES_PVC_STORAGE__CLASS__NAME: "{{ .Values.global.cheWorkspacePVCStorageClassName }}"
+{{- with .Values.che }}
+  {{- with .infra }}
+    {{- with .kubernetes }}
+      {{- with .pvc }}
+        {{- with .jobs }}
+          {{- with .image }}
+  CHE_INFRA_KUBERNETES_PVC_JOBS_IMAGE: {{ . | quote }}
```

Comment:
@tolusha 
I need your opinion for the following.  Since the names of the values are all over the, place, I thought we could simply leverage the property names used internally by `che`.  For example in [PVCSubPathHelper.java](https://github.com/eclipse/che/blob/master/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java#L95)
```
@Named("che.infra.kubernetes.pvc.jobs.image") String jobImage,
@Named("che.infra.kubernetes.pvc.jobs.image.pull_policy") String imagePullPolicy,
```
So no need to be smart and figure out which helm values maps to what.  And it makes it in line with the env variable name too.  The one catch though is you need more code in helm to check for null values.  You need to make that check at every `.` or node.  It might not look too good when you only have 1 value like I have right now, but once you start and group all the values together, you end up with blocks of values...  You can use helm's `with` directive or `if` like I did firther down this file (see line 161 where I set `CHE_SERVER_SECURE__EXPOSER_JWTPROXY_IMAGE`).

It's only a proposition, feel free to reject it and I'll adapt this PR accordingly.

Reply:
That's an interesting question. I don't have answer right now.

Reply:
As far as I understand we will have to following in values.yaml
```
che:
  infra:
    kubernetes:
      pvc:
        jobs:
          image: <..>
  server:
    secureExposer:
      jwtProxyImage: <..>
```

Reply:
I will raise a question in che-dev@eclipse.org mailing list

Reply:
@cccs-eric 

I personally tend to this:
```
che: 
  infraKubernentes:
    pvcJobsImage: <..>
  server:
    secureExposerJwtProxyImage: <..>
```

I will raise a question in che-dev@eclipse.org mailing list

Reply:
I have updated my code with what you proposed.  I had to use a few nested `if` to make sure the tree of values (the parent nodes) exist.  As I said earlier, it would be nice to re-group those under the same `if` to avoid testing the parent nodes each and every time.  But that should go in another PR.

Reply:
After discussion with @sleshchenko  we decided to flatten the fields and follow this pattern in the future.
So, pls update the PR with `cheInfraKubernentesPvcJobsImage` and `cheServerSecureExposerJwtProxyImage`

Reply:
Just a bit more context: Che Server environment variables have autogenerated documentation https://www.eclipse.org/che/docs/che-7/installation-guide/advanced-configuration-options-for-the-che-server-component/
And the idea is to put all Che Server configuration that is propagated to the config map to one field. Then helm template for config map will be simpler: it will just put all items in a loop in Config Map data. We don't need anyone to contribute every single env var into helm chart, users are able just to provision a new value with --value parameters, I suppose.
+ we can reuse the same documentation.

So, I see value.yaml like the following:
```yaml
che:
  customProperties: # name can be different. It's not too custom, so just properties or advanced_cfg, or something else
    CHE_INFRA_KUBERNETES_PVC_JOBS_IMAGE: '...'
   	CHE_SERVER_SECURE__EXPOSER_JWTPROXY_IMAGE: '...'
```

Reply:
Yes, that makes a lot of sense.  It would make the chart aligned with what is going on in the Operator.  It should be fairly trivial to implement.  The only thing we would lose is the possibility of adding logic in the configmap like:
```
if valueX is set, then set valueY
```
but for those cases, the value could come from the chart values (like we are doing now).  But I don't think this would happen often (if it ever does).